arch/arm/include/kernel_arch_func.h

```diff
@@ -42,6 +46,16 @@ _arch_switch_to_main_thread(struct k_thread *main_thread,
 			    k_thread_stack_t *main_stack,
 			    size_t main_stack_size, k_thread_entry_t _main)
 {
+#ifdef CONFIG_ARM_MPU
+	/* Configure static memory map. This will program MPU regions,
+	 * to set up access permissions for fixed memory sections, such
+	 * as Application Memory or No-Cacheable SRAM area.
+	 *
+	 * This function is invoked once, upon system initialization.
+	 */
+	_arch_configure_static_mpu_regions();
```

Comment:
I think that doing that configuring those region in `_arch_switch_to_main_thread()` is a bit late, at least for the `__nocache` region, which has to be setup when the drivers (currently GMAC and MCUX ethernet drivers) are initialized. IIUC the MPU initialization is currently done in PRE_KERNEL_1.

Reply:
Thanks, this is interesting. So, the MPU driver is initialized in PRE_KERNEL_1, correct.

Could you elaborate a bit why the __no_cache region can't wait until kernel/init, which comes before we jump to the _main() (Main thread)? This comes before POST_KERNEL, as far as I can see, anyways.

Otherwise, sure, we could always move this call into the driver itself: Static regions _by design_ do not depend on thread(s)/kernel. But if no big deal, I would prefer to keep the call where it is now :)